### ISSUE TITLE
Types don't work on on `createUseStyles` 

### ISSUE THREAD

**Issue by goleary** [cssinjs/jss#1439]
__Expected behavior:__
I expect the same type inference/autocompletion when using `createUseStyles` that I get when I use the `style` attribute in jsx. Here you can see the autocompletion offered on `<div style...`
![image](https://user-images.githubusercontent.com/16123225/103955638-9a429400-50fb-11eb-9b59-2f04dd94bebf.png)

__Describe the bug:__
There is no autocompletion when using `createUseStyles`
![image](https://user-images.githubusercontent.com/16123225/103955705-bb0ae980-50fb-11eb-81da-5d79b4cd100e.png)

I'm a little bit confused because it appears as though a PR was merged to fix this behaviour but it doesn't seem to be working: https://github.com/cssinjs/jss/pull/1352

@moshest is the behavior you were aiming to fix with that PR? or am I misunderstanding something?

__Codesandbox link:__ 
https://codesandbox.io/s/react-jss-ts-13cqf

__Versions (please complete the following information):__
 - jss: 10.5.0
 - react-jss: 10.5.0
 - react: 17.0.0
Feel free to add any additional versions which you may think are relevant to the bug.

**Comment by moshest** [comment]
Honestly it's hard to follow. A few changes made since this PR and maybe it broken again.

This is the [current status](https://github.com/cssinjs/jss/blob/master/packages/jss/src/index.d.ts):

```ts
export type JssStyle =
  | {
      [K in keyof NormalCssProperties]:
        | NormalCssValues<K>
        | JssStyle
        | Func<NormalCssValues<K> | JssStyle | undefined>
        | Observable<NormalCssValues<K> | JssStyle | undefined>
    }
  | {
      [K: string]:
        | JssValue
        | JssStyle
        | Func<JssValue | JssStyle | undefined>
        | Observable<JssValue | JssStyle | undefined>
    }
```

This is the working [PR version](https://github.com/cssinjs/jss/pull/1352/files):

```ts
export type JssStyle = {
  [K in keyof NormalCssProperties | string]:
    | NormalCssValues<K>
    | JssStyle
    | Func<NormalCssValues<K> | JssStyle | undefined>
 }
```

Can you change the declaration locally and check if that fixes it for you?

**Comment by goleary** [comment]
@moshest when I make the change you suggested I see no change.

I also changed `Styles` to match what it was in your PR:

```
export type Styles<Name extends string | number | symbol = string> = Record<
  Name,
  JssStyle | string | Func<JssStyle | string | null | undefined>
>
```
But it's still not working.

I even updated [my codesandbox](https://codesandbox.io/s/react-jss-ts-13cqf) to use the version where this was supposed to have been fixed  ([10.3.0](https://github.com/cssinjs/jss/releases/tag/v10.3.0)), but I still get no autocomplete.

**Comment by moshest** [comment]
Well it's need further investigating.

I can share that I did this fix on TypeScript 3.x and the attached test file on the PR worked with autocomplete in WebStorm.
I will update if I will find some time to look into that and apply a new fix. Feel free to dive in in the meantime.

**Comment by kof** [comment]
@moshest I have added you to the core team since you have contributed to many typescript changes already, so you can review and merge them as you see fit, feel free to ping me if anything in twitter dm https://twitter.com/kof

**Comment by ymor** [comment]
I believe https://github.com/cssinjs/jss/pull/1423 has fixed this issue and now works correctly in 10.5.1

**Comment by moshest** [comment]
@goleary can you confirm as well?

**Comment by goleary** [comment]
Thanks for the tip @ymor & the ping @moshest 

It is certainly better in the latest release. I am now getting autocomplete on property names.

That said, the traditional react `style` prop provides autocomplete for not only the property names, but also the values:

![image](https://user-images.githubusercontent.com/16123225/107168454-4af8b900-6970-11eb-84bb-dbeb61b63391.png)

That is not working with `createUseStyles`:

![image](https://user-images.githubusercontent.com/16123225/107168608-aaef5f80-6970-11eb-9be7-0d4f2f4f07e2.png)


Was this ever working in `react-jss`?

**Comment by k-yle** [comment]
> Was this ever working in react-jss?

we've being using JSS for 2 years and it's never worked. but I've made a PR that should make this work for TypeScript and JavaScript users: #1454
